modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -225,15 +229,18 @@ export default class Ecdsa {
       let pk = getPaillierPublicKey(n);
       const k = hexToBigInt(shareToBeSend['k']);
 
-      const beta0 = Ecdsa.curve.scalarRandom();
-      shareParticipant.beta = bigIntToBufferBE(Ecdsa.curve.scalarReduce(Ecdsa.curve.scalarNegate(beta0)), 32).toString(
-        'hex'
-      );
+      const beta0 = bigintCryptoUtils.randBetween(n / _3n - _1n);
+      shareParticipant.beta = bigIntToBufferBE(
+        Ecdsa.curve.scalarReduce(Ecdsa.curve.scalarNegate(Ecdsa.curve.scalarReduce(beta0))),
```

Comment:
The extra reduce is not necessary, it's enough to just use `Ecdsa.curve.scalarNegate(Ecdsa.curve.scalarReduce(beta0))`

Reply:
Nice Catch !!, Updated